examples/stm32/xively_demo_stm32f2xx_nucleo_eth/Src/xively_client.c

```diff
@@ -0,0 +1,1214 @@
+
+        if ( t_info->subscribe_cb )
+        {
+            snprintf( fq_topic, XC_TOPIC_LEN, XC_TOPIC_FORMAT, account_id, device_id,
+                      t_info->name );
+
+            printf( "Xively: Subscribe Pending \"%s\".\n", t_info->name );
+            xi_subscribe( ctx, fq_topic, t_info->qos, t_info->subscribe_cb, 0 );
+            e++;
+            xc_subscribe_next_topic = e;
+            break;
+        }
+    }
+}
+
+#if 1 // sensor feature
+
+/* the interval for the time function */
+#define XI_PUBLISH_INTERVAL_SEC 5
+
+typedef struct topic_descr_s
+{
+    const char* const name;
+} mqtt_topic_descr_t;
+
+/* declaration of fn pointer for the msg handler */
+typedef void ( *on_msg_handler_fn_t )( const uint8_t* const msg, size_t msg_size );
+
+/* declaration of fn pointer for the polling push mechanism */
+typedef xi_state_t ( *on_msg_push_fn_t )(
+    const mqtt_topic_descr_t* const mqtt_topic_descr );
+
+/* each topic will have it's descriptor and the handler in case of the message */
+typedef struct mqtt_topic_configuration_s
+{
+    mqtt_topic_descr_t topic_descr;
+    on_msg_handler_fn_t on_msg_pull;
+    on_msg_push_fn_t on_msg_push;
+} mqtt_topic_configuration_t;
+
+/* combines name with the account and device id */
+#define XI_TOPIC_NAME_MANGLE( name )                                                     \
+    "xi/blue/v1/" XI_ACCOUNT_ID "/d/" XI_DEVICE_ID "/" name
+
+/* declaration of topic handlers for SUB'ed topics */
+static void on_led_msg( const uint8_t* const msg, size_t msg_size );
+
+/* declaration of topic handlers for PUB'ed topics */
+static xi_state_t pub_accelerometer( const mqtt_topic_descr_t* const mqtt_topic_descr );
+static xi_state_t pub_magnetometer( const mqtt_topic_descr_t* const mqtt_topic_descr );
+static xi_state_t pub_barometer( const mqtt_topic_descr_t* const mqtt_topic_descr );
+static xi_state_t pub_button( const mqtt_topic_descr_t* const mqtt_topic_descr );
+static xi_state_t pub_gyroscope( const mqtt_topic_descr_t* const mqtt_topic_descr );
+static xi_state_t pub_humidity( const mqtt_topic_descr_t* const mqtt_topic_descr );
+static xi_state_t pub_temperature( const mqtt_topic_descr_t* const mqtt_topic_descr );
+
+/* topic's initialisation function */
+static xi_state_t init_xively_topics( xi_context_handle_t in_context_handle );
+
+/* table of topics used for this demo */
+static const mqtt_topic_configuration_t topics_array[] = {
+    {{XI_TOPIC_NAME_MANGLE( "Accelerometer" )}, NULL, pub_accelerometer},
+    {{XI_TOPIC_NAME_MANGLE( "Magnetometer" )}, NULL, pub_magnetometer},
+    {{XI_TOPIC_NAME_MANGLE( "Barometer" )}, NULL, pub_barometer},
+    {{XI_TOPIC_NAME_MANGLE( "Gyroscope" )}, NULL, pub_gyroscope},
+    {{XI_TOPIC_NAME_MANGLE( "Humidity" )}, NULL, pub_humidity},
+    {{XI_TOPIC_NAME_MANGLE( "Button" )}, NULL, pub_button},
+    {{XI_TOPIC_NAME_MANGLE( "LED" )}, &on_led_msg, NULL},
+    {{XI_TOPIC_NAME_MANGLE( "Temperature" )}, NULL, pub_temperature},
+};

```

Comment:
yeah, I think we should refactor this and use common solution for both projects: eth and wifi. But for now let's keep it duplicated since no resources to do the job. Maybe after the Embedded World Demo?